lib/Transforms/IPO/StructFieldAccessInfo.cpp

```diff
@@ -0,0 +1,349 @@
+// lib/Tranforms/IPO/StructFieldAccessInfo.cpp - Implements class of
+// StructFieldAccessInfo -*- C++ -*-===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===------------------------------------------------------------------------===//
+//
+// This file implements StructFieldAccessInfo that is used in
+// StructFieldCacheAnalysis pass.
+//
+//===------------------------------------------------------------------------===//
+
+#include "StructFieldCacheAnalysisImpl.h"
```

Comment:
Why? I don't know what examples are out there, but unless you have a very strong reason, leave the header file in the include dir.